include/swift/AST/Decl.h

```diff
@@ -5885,7 +5903,53 @@ AbstractStorageDecl::AccessorRecord::getAccessor(AccessorKind kind) const {
   }
   return nullptr;
 }
-  
+
+/// CallDecl - Declares a callable method for a type. For example:
+///
+/// \code
+/// struct Adder {
+///   var base: Int
+///   call(_ x: Int) -> Int {
+///      return base + x
+///   }
+/// }
+/// \endcode
+class CallDecl final : public FuncDecl {
```

Comment:
Conforming `CallDecl` to `FuncDecl` vs `AbstractFunctionDecl` is discussed in [this thread](https://forums.swift.org/t/question-what-should-calldecl-inherit-from/21815?u=dan-zheng).
It has a few nice properties - code sharing with `FuncDecl` is the primary one.

Reply:
Committed to making `CallDecl` inherit from `FuncDecl`.